src/dmd/target.d

```diff
@@ -774,6 +776,37 @@ struct Target
         const sz = t.size(loc);
         return global.params.is64bit ? (sz + 7) & ~7 : (sz + 3) & ~3;
     }
+
+    /**
+     * Get targetInfo by key
+     * Params:
+     *  name = name of targetInfo to get
+     *  loc = location to use for error messages
+     * Returns:
+     *  Expression for the requested targetInfo
+     */
+    extern (C++) static Expression getTargetInfo(const(char)* name, const ref Loc loc)
```

Comment:
Should return a StringExp.

Reply:
No `static` for function declarations.

Reply:
Please, use D strings, not `char*`

Reply:
> No static for function declarations.

All functions are static in this interface.

Reply:
> Should return a StringExp.

It doesn't just return strings.

> No `static` for function declarations.

Every function in `Target` is static.

> Please, use D strings, not `char*`

You specifically didn't want to mangle D strings for C++. I know other people wanted to, but you rejected it. As such, the API's must still use C strings.

Reply:
It returns a StringExp or null, meaning the return type can be StringExp.

Reply:
There’s a follow up PR that returns `IntegerExp` as well.

Reply:
> Should return a StringExp.

No it should _not_, yes _this_ implementation _currently_ returns only strings, but this is not the only implementation. This will create an unnecessary upstream diff for LDC and GDC.

Reply:
Ok, returning Expression is ok.